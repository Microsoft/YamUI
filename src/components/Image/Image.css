/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
@import 'src/css/variables/borders.css';

.y-image {
  display: inline-block;
}

.y-image__fullWidth {
  display: block;
}

.y-image__border-soft {
  border-radius: $borderRadius__soft;
}
.y-image__border-round {
  border-radius: $borderRadius__round;
}

.y-image__block {
  display: block;
}
