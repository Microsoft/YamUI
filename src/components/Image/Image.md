The `Image` component renders an img element within a wrapper div. This div is displayed as inline-block so it behaves like a standard inline image. Height and width actually get applied to the wrapper div, and the image will get scaled/positioned within the wrapper depending on which props you provide.

### Notes for use

Try to set a height and width whenever possible, this will keep the browser from calculating reflows on re-render or on image load.

If an image is used as a supplemental icon, or screenreaders should not read it out loud for any other reason, provide an empty string as `description`. This is a required prop because a missing alt attribute means the screenreader will read the URL out loud.

### Examples

Basic:

```js { "props": { "data-example": "basic" } }
const yammerLogo = 'yammer-logo.png';
const yammerLogoDescription = 'Yammer "y" logo';

<div>
  <Image source={yammerLogo} description={yammerLogoDescription} width={250} height={140} />
</div>
```

Image fit cover:

```js { "props": { "data-example": "fit cover" } }
const { ImageFit } = require('.');

const yammerLogo = 'yammer-logo.png';
const yammerLogoDescription = 'Yammer "y" logo';

<div>
  <p>
    The <strong>cover</strong> option can be used to nicely fill the desired image area while hiding any overflow.
  </p>
  <Image
    source={yammerLogo}
    description={yammerLogoDescription}
    height={250}
    width={250}
    imageFit={ImageFit.cover}
  />
</div>
```

Image fit contain:

```js { "props": { "data-example": "fit contain" } }
const { ImageFit } = require('.');

const yammerLogo = 'yammer-logo.png';
const yammerLogoDescription = 'Yammer "y" logo';

<div>
  <p>
    The <strong>contain</strong> option displays the entire image within our image area and leaves whitespace.
  </p>
  <Image
    source={yammerLogo}
    description={yammerLogoDescription}
    height={250}
    width={250}
    imageFit={ImageFit.contain}
  />
</div>
```

Image fit center:

```js { "props": { "data-example": "fit center" } }
const { ImageFit } = require('.');

const yammerLogo = 'yammer-logo.png';
const yammerLogoDescription = 'Yammer "y" logo';

<div>
  <p>
    The <strong>center</strong> option centers the image horizontally and vertically, using the image's original size and cropping its overflow.
  </p>
  <Image
    source={yammerLogo}
    description={yammerLogoDescription}
    height={250}
    width={250}
    imageFit={ImageFit.center}
  />
</div>
```

Image fit not provided:

```js { "props": { "data-example": "fit not provided" } }
const yammerLogo = 'yammer-logo.png';
const yammerLogoDescription = 'Yammer "y" logo';

<div>
  <p>
    If we don't specify a fit option, the image will get stretched.
  </p>
  <Image source={yammerLogo} description={yammerLogoDescription} height={250} width={250} />
</div>
```

Image fit none:

```js { "props": { "data-example": "fit none" } }
const { ImageFit } = require('.');

const yammerLogo = 'yammer-logo.png';
const yammerLogoDescription = 'Yammer "y" logo';

<div>
  <p>
    The <strong>none</strong> option places the image at the top left of its <strong>div</strong> wrapper. It will render at its original size, possibly showing whitespace or hiding its overflow. This is generally not an option you will want to use.
  </p>
  <Image
    source={yammerLogo}
    description={yammerLogoDescription}
    height={250}
    width={250}
    imageFit={ImageFit.none}
  />
</div>
```

Full width:

```js { "props": { "data-example": "full width" } }
const { ImageFit } = require('.');
const { FixedGridColumn, FixedGridRow, GutterSize } = require('../FixedGrid')

const yammerLogo = 'yammer-logo.png';
const yammerLogoDescription = 'Yammer "y" logo';

const fullWidthImage = (
  <Image source={yammerLogo} description={yammerLogoDescription} fullWidth />
);

<div style={{ maxWidth: '500px' }}>
  <FixedGridRow bottomSpacing={GutterSize.SMALL}>
    Your image will grow to fit the width of its container. Use this option when you have a grid or other layout component determining the space an image should fill. Any provided height and width will be ignored in this case, replaced with width: 100% and height: auto.
  </FixedGridRow>
  <FixedGridRow bottomSpacing={GutterSize.SMALL}>
    <FixedGridColumn>{fullWidthImage}</FixedGridColumn>
    <FixedGridColumn>{fullWidthImage}</FixedGridColumn>
    <FixedGridColumn>{fullWidthImage}</FixedGridColumn>
    <FixedGridColumn>{fullWidthImage}</FixedGridColumn>
  </FixedGridRow>
  <FixedGridRow bottomSpacing={GutterSize.SMALL}>
    <FixedGridColumn>{fullWidthImage}</FixedGridColumn>
    <FixedGridColumn>{fullWidthImage}</FixedGridColumn>
  </FixedGridRow>
  <FixedGridRow bottomSpacing={GutterSize.SMALL}>
    <FixedGridColumn>{fullWidthImage}</FixedGridColumn>
  </FixedGridRow>
</div>
```

Handling load state:

```js { "props": { "data-example": "load state" } }
const { ImageFit, ImageLoadState } = require('.');

const badImage = '404.png';
const yammerLogoOrange = 'yammer-logo-orange.jpg';
const yammerLogoDescription = 'Yammer "y" logo';

class StateChangeDemo extends React.PureComponent {
  constructor() {
    super();
    this.state = { source: badImage };
    this.handleStateChange = this.handleStateChange.bind(this);
  }

  render() {
    return (
      <Image
        source={this.state.source}
        description={yammerLogoDescription}
        height={150}
        width={250}
        imageFit={ImageFit.cover}
        onLoadingStateChange={this.handleStateChange}
      />
    );
  }

  handleStateChange(loadState) {
    if (loadState === ImageLoadState.error) {
      this.setState({ source: yammerLogoOrange });
    }
  }
}

<div>
  <p>
    When an image's loading state changes, it will trigger the <em>onLoadingStateChange</em> callback. This example updates the source URL passed to the image when its <em>onLoadingStateChange</em> is called with an error.
  </p>
  <StateChangeDemo />
</div>
```