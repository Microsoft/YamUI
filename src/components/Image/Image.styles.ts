/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { borders } from '../../util/styles/borders';
import { mergeStyleSets } from 'office-ui-fabric-react/lib/Styling';
import { memoizeFunction } from 'office-ui-fabric-react/lib/Utilities';
import { BorderType } from './';

export interface BoxStyleProps {
  borderType?: BorderType;
  fullWidth?: boolean;
  block?: boolean;
}

export const getClassNames = memoizeFunction((styleProps: BoxStyleProps) => {
  const { borderType, fullWidth, block } = styleProps;

  return mergeStyleSets({
    root: {
      display: fullWidth || block ? 'block' : 'inline-block',
      borderRadius: borderType ? borders[borderType] : undefined,
    },
  });
});
