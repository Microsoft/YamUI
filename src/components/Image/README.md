# Image

This component renders an image with some convenient ways to manage aspect ratio and overflow.

## Properties

| Prop | Type | Description |
| ---- | ---- | ----------- |
| source * | string | Image URL |
| description * | string | The imge's alt text, read aloud by screenreaders and visually rendered by the browser when the image fails to load. |
| height | number | The number of pixels tall the image should be rendered at. Leave empty to auto-scale. |
| width | number | The number of pixels wide the image should be rendered at. Leave empty to auto-scale. |
| imageFit | ImageFit | How to fit the image within the given dimensions. See the [Fabric Image docs](https://dev.office.com/fabric#/components/image) for more info. |
| fullWidth | boolean | If true, height and width will be ignored and the image will grow to 100% wide and auto height (maintaining its aspect ratio). |
| onLoadingStateChange | (loadState: ImageLoadState) => void | Optional callback handler which will receive a Fabric ImageLoadState ('error' | 'loaded' | 'notLoaded'). |

\* indicates property is required

## Notes for use

Try to set a height and width whenever possible, this will keep the browser from calculating reflows on re-render or on image load.

If an image is used as a supplemental icon, or screenreaders should not read it out loud for any other reason, provide an empty string as `description`. This is a required prop because a missing alt attribute means the screenreader will read the URL out loud.

## Examples

### Basic usage

```js
<Image
  source="my_image.png"
  description="My image description"
  height={30}
  width={40}
/>
```

### Reponsive full-width image maintaining aspect ratio

```js
<Image
  source="my_image.png"
  description="My image description"
  fullWidth
/>
```

### Cropping a rectangular image to fill square dimensions

```js
<Image
  source="100x200.png"
  description="Rectangle cropped to thumbnail"
  height={100}
  width={100}
  imageFit={ImageFit.cover}
/>
```
