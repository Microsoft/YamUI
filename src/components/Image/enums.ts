/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum BorderType {
  SOFT = 'soft',
  ROUND = 'round',
}
