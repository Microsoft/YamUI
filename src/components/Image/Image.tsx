/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { join } from '../../util/classNames';
import { Image as FabricImage, ImageFit, ImageLoadState } from 'office-ui-fabric-react/lib/Image';
import { BaseComponentProps } from '../../util/BaseComponent/props';
import { BorderType } from './types';
import { getClassNames } from './Image.styles';
export { BorderType, ImageFit, ImageLoadState };

export interface ImageProps extends BaseComponentProps {
  /**
   * Whether the clickable should be `display: block`.
   */
  block?: boolean;

  /**
   * Description of the image that must be provided for screenreaders. Set to an empty string if
   * screenreaders should not read anything out loud.
   */
  description: string;

  /**
   * Source URL.
   */
  source: string;

  /**
   * Height in pixels.
   */
  height?: number;

  /**
   * Width in pixels.
   */
  width?: number;

  /**
   * Set this to true in responsive layouts where the image's container element should dictate the
   * image's width. The image will grow to fit its container's width and its height will grow to
   * match its natural aspect ratio.
   */
  fullWidth?: boolean;

  /**
   * Specifies how to position this image within its div wrapper. Not setting a value will behave
   * as expected, setting the image's height and width to the provides values and stretching/skewing
   * the image if its actual dimensions don't match. Providing a value allows you to specify a
   * different height/width than the image's actual dimensions while covering the area, containing
   * the image within the area, centering the image without scaling or stretching, etc.
   */
  imageFit?: ImageFit;

  /**
   * Image can have the default square corners, a soft 2px radius, or a fully round border.
   */
  borderType?: BorderType;

  /**
   * Will hide the image until it has loaded, then fade it in.
   * @default false
   */
  shouldFadeIn?: boolean;

  /**
   * Callback to be invoked when the image's loading state changes.
   */
  onLoadingStateChange?: ((loadState: ImageLoadState) => void);
}

/**
 * An `Image` renders an `img` element within a wrapper `div`. This `div` is displayed as
 * `inline-block`, so it behaves like a standard inline image. The `height` and `width` actually get
 * applied to the wrapper `div`, and the image will get scaled/positioned within the wrapper
 * depending on which props you provide.
 */
export default class Image extends React.Component<ImageProps> {
  public static defaultProps: Partial<ImageProps> = {
    fullWidth: false,
    shouldFadeIn: false,
  };

  public render() {
    const {
      description,
      fullWidth,
      height,
      width,
      imageFit,
      source,
      shouldFadeIn,
      onLoadingStateChange,
      borderType,
      block,
      className,
    } = this.props;
    const classNames = getClassNames({ borderType, fullWidth, block });
    return (
      <FabricImage
        alt={description}
        className={join(['y-image', className, classNames.root])}
        height={height}
        imageFit={imageFit}
        width={fullWidth ? '100%' : width}
        src={source}
        onLoadingStateChange={onLoadingStateChange}
        shouldFadeIn={shouldFadeIn}
      />
    );
  }
}
