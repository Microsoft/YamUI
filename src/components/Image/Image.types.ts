/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { BaseComponentProps } from '../../util/BaseComponent/props';
import { ImageFit, ImageLoadState } from 'office-ui-fabric-react/lib/Image';
export { ImageFit, ImageLoadState };

export interface ImageProps extends BaseComponentProps {
  /**
   * Whether the clickable should be `display: block`.
   */
  block?: boolean;

  /**
   * Description of the image that must be provided for screenreaders. Set to an empty string if
   * screenreaders should not read anything out loud.
   */
  description: string;

  /**
   * Source URL.
   */
  source: string;

  /**
   * Height in pixels.
   */
  height?: number;

  /**
   * Width in pixels.
   */
  width?: number;

  /**
   * Set this to true in responsive layouts where the image's container element should dictate the
   * image's width. The image will grow to fit its container's width and its height will grow to
   * match its natural aspect ratio.
   */
  fullWidth?: boolean;

  /**
   * Specifies how to position this image within its div wrapper. Not setting a value will behave
   * as expected, setting the image's height and width to the provides values and stretching/skewing
   * the image if its actual dimensions don't match. Providing a value allows you to specify a
   * different height/width than the image's actual dimensions while covering the area, containing
   * the image within the area, centering the image without scaling or stretching, etc.
   */
  imageFit?: ImageFit;

  /**
   * Image can have the default square corners, a soft 2px radius, or a fully round border.
   */
  borderType?: BorderType;

  /**
   * Will hide the image until it has loaded, then fade it in.
   * @default false
   */
  shouldFadeIn?: boolean;

  /**
   * Callback to be invoked when the image's loading state changes.
   */
  onLoadingStateChange?: ((loadState: ImageLoadState) => void);
}

export enum BorderType {
  SOFT = 'soft',
  ROUND = 'round',
}
