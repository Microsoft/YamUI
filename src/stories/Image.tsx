/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Image, { ImageFit, ImageLoadState } from '../components/Image';
import withReadme from 'storybook-readme/with-readme';
const readme = require('../components/Image/README');


const yammerLogo = require('./assets/img/yammer-logo.png');
const yammerLogoOrange = require('./assets/img/yammer-logo-orange.jpg');
const yammerLogoDescription = 'Yammer "y" logo';

storiesOf('Image', module)
  .addDecorator(withReadme(readme))
  .add('basic image', () => (
    <div>
      The Image component renders an img element within a wrapper div. This div is displayed as inline-block
      so it behaves like a standard inline image. Height and width actually get applied to the wrapper div,
      and the image will get scaled/positioned within the wrapper depending on which props you provide.
      <br />
      <Image source={yammerLogo}
             description={yammerLogoDescription}
             width={250}
             height={140} />
    </div>
  ))
  .add('image fit: not provided', () => (
    <div>
      Our example image is a rectangle, so if we provide equal height and width without providing
      an imageFit prop it gets stretched:
      <br />
      <Image source={yammerLogo}
             description={yammerLogoDescription}
             height={250}
             width={250} />
    </div>
  ))
  .add('image fit: cover', () => (
    <div>
      Here we set imageFit: ImageFit.cover. This can be used to nicely fill the
      desired image area while hiding any overflow:
      <br />
      <Image source={yammerLogo}
             description={yammerLogoDescription}
             height={250}
             width={250}
             imageFit={ImageFit.cover} />
    </div>
  ))
  .add('image fit: contain', () => (
    <div>
      Here we set imageFit: ImageFit.contain, which displays the entire image within our
      image area and leaves whitespace:
      <br />
      <Image source={yammerLogo}
             description={yammerLogoDescription}
             height={250}
             width={250}
             imageFit={ImageFit.contain} />
    </div>
  ))
  .add('image fit: center', () => (
    <div>
      Here we set imageFit: ImageFit.center. This centers the image horizontally and vertically,
      using the image's original size and cropping its overflow.
      <br />
      <Image source={yammerLogo}
             description={yammerLogoDescription}
             height={250}
             width={250}
             imageFit={ImageFit.center} />
      <br />
    </div>
  ))
  .add('image fit: none', () => (
    <div>
      Here we set imageFit: ImageFit.none. This places the image at the top left of its div
      wrapper. It will render at its original size, possibly showing whitespace or hiding its overflow.
      This is generally not an option you will want to use.
      <br />
      <Image source={yammerLogo}
             description={yammerLogoDescription}
             height={250}
             width={250}
             imageFit={ImageFit.none} />
      <br />
    </div>
  ))
  .add('full width', () => {
    // TODO: Clean this up when we have grid components
    const fullWidthImage = <Image source={yammerLogo} description={yammerLogoDescription} fullWidth={true} />;
    return(
      <div>
        <table style={{ maxWidth: '500px' }}>
          <tbody>
            <tr>
              <td colSpan={4}>
                If you set fullWidth: true your image will grow to fit the width of its container. Use this option when
                you have a grid or some other layout component determining the space an image should fill.
                Any provided height and width will be ignored in this case, replaced with width: 100% and height: auto.
              </td>
            </tr>
            <tr>
              <td style={{ width: '25%' }}>{fullWidthImage}</td>
              <td style={{ width: '25%' }}>{fullWidthImage}</td>
              <td style={{ width: '25%' }}>{fullWidthImage}</td>
              <td style={{ width: '25%' }}>{fullWidthImage}</td>
            </tr>
            <tr>
              <td colSpan={2} style={{ width: '50%' }}>{fullWidthImage}</td>
              <td colSpan={2} style={{ width: '50%' }}>{fullWidthImage}</td>
            </tr>
            <tr>
              <td colSpan={4}>{fullWidthImage}</td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  })
  .add('load events', () => {
    interface StateChangeDemoState {
      source: string;
    }
    class StateChangeDemo extends React.Component<{}, StateChangeDemoState> {
      constructor () {
        super();
        this.state = {
          source: '404.png',
        };

        this.handleStateChange = this.handleStateChange.bind(this);
      }
      render () {
        return (
          <Image source={this.state.source}
                 description={yammerLogoDescription}
                 height={150}
                 width={250}
                 imageFit={ImageFit.cover}
                 onLoadingStateChange={this.handleStateChange} />
        );
      }
      handleStateChange (loadState: ImageLoadState) {
        action(ImageLoadState[loadState])();
        if (loadState === ImageLoadState.error) {
          this.setState({ source: yammerLogoOrange });
        }
      }
    }
    return (
      <div>
        When an Image component's loading state changes it will trigger the onLoadingStateChange prop
        callback, providing a loading state of enum ImageLoadState. This example updates the source URL
        passed to the Image component when its onLoadingStateChange is called with ImageLoadState.error.
        <br />
        <StateChangeDemo />
      </div>
    );
  });
