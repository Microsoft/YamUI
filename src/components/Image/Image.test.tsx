/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';
import Image, { BorderType, ImageProps, ImageFit, ImageLoadState } from '.';

describe('<Image />', () => {
  let component: ShallowWrapper<ImageProps>;
  let fullComponent: ReactWrapper<ImageProps>;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<Image source="image.png" description="description" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<Image source="image.png" description="description" className="TEST_CLASSNAME" />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with height 50 and width 100', () => {
    beforeEach(() => {
      component = shallow(<Image source="image.png" description="description" height={50} width={100} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with imageFit cover', () => {
    beforeEach(() => {
      component = shallow(<Image source="image.png" description="description" imageFit={ImageFit.cover} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with border type', () => {
    beforeEach(() => {
      component = shallow(<Image source="image.png" description="description" borderType={BorderType.ROUND} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with fullWidth', () => {
    beforeEach(() => {
      component = shallow(<Image source="image.png" description="description" fullWidth={true} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with shouldFadeIn', () => {
    beforeEach(() => {
      component = shallow(<Image source="image.png" description="description" shouldFadeIn={true} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with block', () => {
    beforeEach(() => {
      component = shallow(<Image source="image.png" description="description" block={true} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when provided a loading state callback', () => {
    let loadState: ImageLoadState | undefined;
    function setLoadState(newLoadState: ImageLoadState) {
      loadState = newLoadState;
    }

    beforeEach(() => {
      jest.useFakeTimers();
      loadState = undefined;
      fullComponent = mount(<Image source="image.png" description="description" onLoadingStateChange={setLoadState} />);
    });

    it('triggers a callback with loaded status on load', () => {
      fullComponent.find('img').simulate('load');
      setTimeout(() => {}, 1);

      jest.runAllTimers();
      expect(loadState).toEqual(ImageLoadState.loaded);
    });

    it('triggers a callback with error status on error', () => {
      fullComponent.find('img').simulate('error');
      setTimeout(() => {}, 1);

      jest.runAllTimers();
      expect(loadState).toEqual(ImageLoadState.error);
    });
  });
});
