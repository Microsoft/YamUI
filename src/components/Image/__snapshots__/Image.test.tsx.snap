// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> with additional className matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      TEST_CLASSNAME
      {
        display: inline-block;
      }
  shouldFadeIn={false}
  src="image.png"
/>
`;

exports[`<Image /> with block matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      {
        display: block;
      }
  shouldFadeIn={false}
  src="image.png"
/>
`;

exports[`<Image /> with border type matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      {
        border-radius: 50%;
        display: inline-block;
      }
  shouldFadeIn={false}
  src="image.png"
/>
`;

exports[`<Image /> with default options matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      {
        display: inline-block;
      }
  shouldFadeIn={false}
  src="image.png"
/>
`;

exports[`<Image /> with fullWidth and height matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      {
        display: block;
      }
  height={50}
  shouldFadeIn={false}
  src="image.png"
  width="100%"
/>
`;

exports[`<Image /> with fullWidth matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      {
        display: block;
      }
  shouldFadeIn={false}
  src="image.png"
  width="100%"
/>
`;

exports[`<Image /> with height 50 and width 100 matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      {
        display: inline-block;
      }
  height={50}
  shouldFadeIn={false}
  src="image.png"
  width={100}
/>
`;

exports[`<Image /> with imageFit cover matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      {
        display: inline-block;
      }
  imageFit={2}
  shouldFadeIn={false}
  src="image.png"
/>
`;

exports[`<Image /> with shouldFadeIn matches its snapshot 1`] = `
<StyledCustomizedImage
  alt="description"
  className=
      y-image
      {
        display: inline-block;
      }
  shouldFadeIn={true}
  src="image.png"
/>
`;
